src/Symfony/Component/HttpKernel/EventListener/PartialDenormalizationExceptionListener.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Serializer\Exception\PartialDenormalizationException;
+use Symfony\Component\Translation\IdentityTranslator;
+use Symfony\Component\Validator\ConstraintViolation;
+use Symfony\Component\Validator\ConstraintViolationList;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+final class PartialDenormalizationExceptionListener implements EventSubscriberInterface
+{
+    public function __construct(private ?TranslatorInterface $translator = null)
+    {
+        if (!$this->translator) {
+            $this->translator = new IdentityTranslator();
+        }
+    }
```

Comment:
```suggestion
    public function __construct(
        private ?TranslatorInterface $translator = new IdentityTranslator(),
    ) {
    }
```

Reply:
I've try this at the beginning but sadly DependencyInjection component explicitly provides `null` even if I'm using `ignoreOnInvalid` behavior :(

Reply:
In that case the alternative can be `$this->translator ??= new IdentityTranslator()`